iocore/net/SSLNetVConnection.cc

```diff
@@ -1095,6 +1105,12 @@ SSLNetVConnection::sslServerHandShakeEvent(int &err)
   }
 
   ssl_error_t ssl_error = SSLAccept(ssl);
+  // if non SNI-Hook set before or SSL Session reuse here:
+  //   the sslHandshakeHookState should be HOOKS_CERT after SSLAccept().
+  // thus, set it to HOOKS_DONE directly.
+  if (HANDSHAKE_HOOKS_CERT == sslHandshakeHookState) {
+    sslHandshakeHookState = HANDSHAKE_HOOKS_DONE;
+  }
```

Comment:
please hold this pull request，I found a bug on it，and will release a new patch.
set HookState to DONE if SSLAccept() only send a Server Hello and without certificate.
next time, SSLAccept() send certificate to client but the HookState already set to DONE. the default certificate will be used and no Cert Hooks called. I will be update patch soon.

sorry ...